secret.py

```
@@ -0,0 +1,14 @@
+def log_in(password):
+    conn = sqlite3.connect('users.db')
+    cursor = conn.cursor()
+
+    hashed_password = hash_password(password)
+    
+    query = f"SELECT * FROM users WHERE username = '{username}' AND password = '{hashed_password}'"
```

Ruff (0.8.2)

Undefined name `username`

(F821)